src/modules/transaction/core/use-cases/withdraw.use-case.ts

```
@@ -0,0 +1,21 @@
+import { OrderInput } from '../../api/graphql/input-types/order-input';
+import { DRWithdrawUseCase } from './dr-withdraw.use-case';
+import { MXWithdrawUseCase } from './mx-withdraw.use-case';
+
+export interface WithdrawUseCaseDto {
+  input: OrderInput;
+  userId: string;
+}
+
+export class WithdrawUseCase {
+  constructor(
+    private readonly drWithdrawUseCase: DRWithdrawUseCase,
+    private readonly mxWithdrawUseCase: MXWithdrawUseCase
+  ) {}
+
+  public async execute(input: WithdrawUseCaseDto) {
+    const { from_currency, to_currency } = input.input;
+    if (from_currency === 'MXN' || to_currency === 'MXN') return await this.mxWithdrawUseCase.execute(input);
+    if (from_currency === 'DOP' || to_currency === 'DOP') return await this.drWithdrawUseCase.execute(input);
+  }
```

This should throw an error if no valid currency is provided. Instead of having hardcoded strings, we should have a `SupportedCurrency` enum to describe all the possible values for a currency that we currently support.

```typescript
export enum SupportedCurrency {
    MXN = "MXN",
    DOP = "DOP
}
```

Don't forget to add types everywhere as looking at the code as it is, it is not clear what the return type of this use case is and we also lose type safety